probe-rs/src/core/registers.rs

```diff
@@ -440,39 +489,52 @@ impl CoreRegisters {
     pub fn msp(&self) -> Option<&CoreRegister> {
         self.0
             .iter()
-            .find(|r| r.role == Some(RegisterRole::MainStackPointer))
+            .find(|r| r.register_has_role(RegisterRole::MainStackPointer))
             .cloned()
     }
 
     /// The process stack pointer.
     pub fn psp(&self) -> Option<&CoreRegister> {
         self.0
             .iter()
-            .find(|r| r.role == Some(RegisterRole::ProcessStackPointer))
+            .find(|r| r.register_has_role(RegisterRole::ProcessStackPointer))
             .cloned()
     }
 
     /// The processor status register.
     pub fn psr(&self) -> Option<&CoreRegister> {
         self.0
             .iter()
-            .find(|r| r.role == Some(RegisterRole::ProcessorStatus))
+            .find(|r| r.register_has_role(RegisterRole::ProcessorStatus))
             .cloned()
     }
 
-    /// Other architecture specific registers
+    /// Return an iterator over all the architecture specific registers which do not have a specific role.
     pub fn other(&self) -> impl Iterator<Item = &CoreRegister> {
         self.0
             .iter()
-            .filter(|r| matches!(r.role, Some(RegisterRole::Other(_))))
+            .filter(|r| {
+                if let Some(role_list) = r.roles {
+                    let mut is_other_register = false;
+                    for role in role_list {
+                        if matches!(role, RegisterRole::Other(_)) {
+                            is_other_register = true;
+                            break;
+                        }
+                    }
+                    is_other_register
+                } else {
+                    false
+                }
+            })
             .cloned()
     }
 
     /// Find an architecture specific register by name
-    pub fn other_by_name(&self, name: &str) -> Result<&CoreRegister, Error> {
+    pub fn other_by_name(&self, name: &'static str) -> Result<&CoreRegister, Error> {
```

Comment:
Please remove the `'static` lifetime here, it should be possible to get register also with non-static strs.

Reply:
@Tiwalun, if I don't have the `'static`, then I get the following.

```
error: lifetime may not live long enough
   --> probe-rs/src/core/registers.rs:515:63
    |
512 |     pub fn other_by_name(&self, name: &str) -> Result<&CoreRegister, Error> {
    |                                       - let's call the lifetime of this reference `'1`
...
515 |             .find(|r| r.register_has_role(RegisterRole::Other(name)))
    |                                                               ^^^^ this usage requires that `'1` must outlive `'static`
    ```

    Is there an easy way to overcome this?

Reply:
You just need to change the thing on register to be String instead of &'static str :)

Reply:
This should work:

```rust
        self.0
            .iter()
            .find(|r| {
                r.roles
                    .iter()
                    .any(|role| matches!(role, RegisterRole::Other(n) if *n == name))
            })
            .ok_or_else(|| Error::GenericCoreError(format!("Argument register {name:?} not found")))
            .cloned()
```

But I've now seen as well that the error message looks wrong, and I feel that an `Option` would be suited better here anyways.

Reply:
Then I get ...
```
error[E0015]: cannot call non-const fn `<str as std::string::ToString>::to_string` in statics
  --> probe-rs/src/architecture/arm/core/registers/aarch64.rs:96:43
   |
96 |         roles: &[RegisterRole::Other("XR".to_string())],
   |                                           ^^^^^^^^^^^
   |
   = note: calls in statics are limited to constant functions, tuple structs and tuple variants
   = note: consider wrapping this expression in `Lazy::new(|| ...)` from the `once_cell` crate: https://crates.io/crates/once_cell
```

Reply:
> This should work:

Yes, it does :) Thanks.